library/packages/src/lib/packages/repository.rb

```diff
@@ -0,0 +1,118 @@
+require "packages/repository_product"
+
+module Packages
+  class Repository
+    Yast.import "Pkg"
+    Yast.import "URL"
+    
+    # @return [Fixnum] Repository ID
+    attr_reader :repo_id
+    # @return [String] Repository name
+    attr_reader :name
+    # @return [String] Repository URL
+    attr_reader :url
+
+    # The repository scheme could not be determined
+    class UnknownScheme < StandardError; end
+
+    # Repository was not found
+    class NotFound < StandardError; end
+
+    class << self
+      # Return all registered repositories
+      #
+      # @see Yast::Pkg.SourceGetCurrent
+      # @see Packages::Repository.find
+      def all
+        Yast::Pkg.SourceGetCurrent(true).map do |repo_id|
+          find(repo_id)
+        end
+      end
+
+      # Return a repository with the given repo_id
+      #
+      # @param repo_id [Fixnum] Repository ID
+      # @return [Packages::Repository] Repository instance
+      #
+      # @raise NotFound
+      def find(repo_id)
+        repo_data = Yast::Pkg.SourceGeneralData(repo_id)
+        raise NotFound if repo_data.nil?
+        new(repo_id: repo_id, enabled: repo_data["enabled"],
+          name: repo_data["name"], autorefresh: repo_data["autorefresh"],
+          url: repo_data["url"])
+      end
+    end
+    
+    # Constructor
+    #
+    # @param repo_id     [Fixnum]  Repository ID
+    # @param name        [String]  Name
+    # @param enabled     [Boolean] Is the repository enabled?
+    # @param autorefresh [Boolean] Is auto-refresh enabled for this repository?
+    # @param url         [String]  Repository URL
+    def initialize(repo_id:, name:, enabled:, autorefresh:, url:)
+      @repo_id = repo_id
+      @name    = name
+      @enabled = enabled
+      @autorefresh = autorefresh
+      @url     = url
+    end
+
+    # Return repository scheme
+    #
+    # The scheme is determined using the URL
+    #
+    # @raise UnknownScheme
+    def scheme
+      return @scheme unless @scheme.nil?
+      data_from_url = Yast::URL.Parse(url)
```

Comment:
Use the native Ruby `URI` class if you do not need any `Yast::URL` specific feature (which is not the case here).